hasjob/models/jobpost.py

```diff
@@ -64,6 +64,13 @@ def starred_job_ids(user, agelimit):
 User.starred_job_ids = starred_job_ids
 
 
+def has_starred_post(user, post):
+    """Checks if user has starred a particular post"""
+    return post.id in user.starred_job_ids() if post else False
```

Comment:
This is going to be inefficient if the user has starred a lot of jobs. Maybe we should eliminate that load entirely and just check for the user's status with the given jobpost only.